### ISSUE TITLE
Playing an animation another time in AnimatedSprite doesn't work

### ISSUE THREAD

**Issue by dploeger** [godotengine/godot#55068]
### Godot version

3.3.4

### System information

macOS, Windows

### Issue description

When you have an AnimatedSprite and use GDScript to play an animation, wait for it to complete and then play the same animation again, it doesn't play it.

### Steps to reproduce

* Add an AnimatedSprite node with frames and an non-looped animation
* Use GDScript to run $AnimatedSprite.play(animation)
* Yield for "animation_finished"
* Use GDScript to run $AnimatedSprite.play(animation) again

### Minimal reproduction project

[testdoubleanim.zip](https://github.com/godotengine/godot/files/7558132/testdoubleanim.zip)

**Comment by Giwayume** [comment]
Try this?
```
$AnimatedSprite.play("default")
yield($AnimatedSprite, "animation_finished")
$AnimatedSprite.stop()
$AnimatedSprite.play("default")
yield($AnimatedSprite,"animation_finished")
```

From what I understand that signal is emitted at the time the last frame is drawn, not necessarily when the animation has already stopped playing. If you try to play an animation that is currently playing, it is a noop.

**Comment by dploeger** [comment]
@Giwayume Nope. Doesn't work. I tried that in the example.

**Comment by Torguen** [comment]
I think the timeline stays at the end and the user should bring it back to 0 with seek

**Comment by dploeger** [comment]
@Torguen You're right. Not seek with AnimatedSprite, but setting frame to 0 works.

However, this is quite confusing because even playing an animation backwards requires the frame to be set to 0 afterwards (which it already is)

So this at least is a documentation issue, though I'd expect this to happen in the stop command.

**Comment by Giwayume** [comment]
Oh right, it says in the documentation `stop()` does not reset the frame counter. So `stop()` is basically `pause()` lol. It is a little bit weird API.
